server/instrumentation-backend/src/sh/calaba/instrumentationbackend/actions/location/FakeGPSLocation.java

```diff
@@ -46,12 +46,6 @@ public Result execute(String... args) {
             return Result.failedResult(e.getMessage());
         }
 
-        if (Build.VERSION.SDK_INT <= 22) {
-            if (context.checkCallingOrSelfPermission(Manifest.permission.ACCESS_MOCK_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-                return Result.failedResult("The application does not have access mock location permission. Add the permission '" + Manifest.permission.ACCESS_MOCK_LOCATION + "' to your manifest");
-            }
-        }
-
```

Comment:
Are you 100% sure we can remove this check? I see you added the permission to the manifest - but maybe really old devices has some quirky behaviour wrt mock locations....

Reply:
I will double check this. Anyway I remember that I had gradle exception when I tried to build this

Reply:
I can return it back. This will work in debug configuration.